src/helper/class-ee-site.php

```diff
@@ -1992,6 +1996,204 @@ protected function custom_site_ssl() {
 		$this->fs->copy( $this->site_data['ssl_crt'], $ssl_crt_dest, true );
 	}
 
+	/**
+	 * Syncs a website from source to a new website in destination.
+	 *
+	 * Note: SSL needs to be added explicitly in the new site.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <source>
+	 * : Name of source website to be cloned. Format [user@ssh-hostname:]sitename
+	 *
+	 * <destination>
+	 * : Name of destination website to be cloned. Format [user@ssh-hostname:]sitename
+	 *
+	 * [--files]
+	 * : Sync only files.
+	 *
+	 * [--db]
+	 * : Sync only database.
+	 *
+	 * [--uploads]
+	 * : Sync only uploads.
+	 *
+	 * [--ssl]
+	 * : Enables ssl on site.
+	 * ---
+	 * options:
+	 *      - le
+	 *      - off
+	 *      - self
+	 *      - inherit
+	 *      - custom
+	 * ---
+	 *
+	 * [--ssl-key=<ssl-key-path>]
+	 * : Path to the SSL key file.
+	 *
+	 * [--ssl-crt=<ssl-crt-path>]
+	 * : Path to the SSL crt file.
+	 *
+	 * [--wildcard]
+	 * : Gets wildcard SSL.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Clone site on same host
+	 *     $ ee site clone foo.com bar.com
+	 *
+	 *     # Clone site from remote server
+	 *     $ ee site clone root@foo.com:foo.com bar.com
+	 *
+	 *     # Clone site from remote with same name
+	 *     $ ee site clone root@foo.com:foo.com .
+	 *
+	 *     # Clone site to remote
+	 *     $ ee site clone foo.com root@foo.com:bar.com
+	 *
+	 *     # Clone site from remote. Only clone files and db, not uploads
+	 *     $ ee site clone root@foo.com:foo.com bar.com --files --db
+	 *
+	 *     # Clone site from remote and enable ssl
+	 *     $ ee site clone root@foo.com:foo.com bar.com --ssl=le --wildcard
+	 *
+	 */
+	public function clone( $args, $assoc_args ) {
+
+		list( $source, $destination ) = get_transfer_details( $args[0], $args[1] );
+
+		try {
+
+			check_site_access( $source, $destination, $assoc_args );
+
+			if ( get_flag_value( $assoc_args, 'files' ) ) {
+				$sync = 'files';
+			} elseif ( get_flag_value( $assoc_args, 'uploads' ) ) {
+				$sync = 'uploads';
+			} elseif ( get_flag_value( $assoc_args, 'db' ) ) {
+				$sync = 'db';
+			} else {
+				$sync = 'all';
+			}
+
+			$operations = [
+				'sync' =>  $sync
+			];
+
+			if ( $destination->create_site( $source, $assoc_args )->return_code ) {
+				EE::error( 'Cannot create site ' . $destination->name . '. Please check logs for more info or rerun the command with --debug flag.' );
+			} else {
+				$destination->ensure_site_exists();
+				$destination->set_site_details();
+			}
+
+			if ( $operations['sync'] === 'files' || $operations['sync'] === 'uploads' || $operations['sync'] === 'all' ) {
+				EE::log( 'Syncing files' );
+				copy_site_files( $source, $destination, $operations['sync'] );
+			}
+
+			if ( $operations['sync'] === 'db' || $operations['sync'] === 'all' ) {
+				EE::log( 'Syncing database' );
+				copy_site_db( $source, $destination );
+			}
+			EE::runcommand( ' site info ' . $destination->name );
```

Comment:
This command always runs on the machine from where clone command is being executed. 
Leading to:

1. Wrong site info if site name is same on both servers.
2. Error in case site being cloned on remote server does not exist locally.